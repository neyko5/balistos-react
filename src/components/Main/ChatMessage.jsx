// @flow

import React from 'react';
import styled from 'styled-components';

const Message = styled.div`
  margin-bottom: 3px;
  color: #3e414c;
  font-size: 12px;
`;
export const Author = styled.span`
  margin-right: 3px;
  font-weight: 700;
  color: ${props => (props.current ? '#9FA600' : '#3e414c')}
`;

const ChatMessage = (props: {
  username: string,
  message: any,
}) => (
  <Message>
    <Author
      current={props.message.user.username === props.username}
      title={new Date(props.message.createdAt).toDateString()}
    >{props.message.user.username}:
    </Author>
    <span>{props.message.message}</span>
  </Message>
);

export default ChatMessage;
