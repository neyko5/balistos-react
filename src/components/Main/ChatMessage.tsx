import React from 'react';
import styled from 'styled-components';
import { ChatMessageType } from '../../types/index';

const Message = styled.div`
    margin-bottom: 3px;
    color: #3e414c;
    font-size: 12px;
`;
export const Author = styled.span`
    margin-right: 3px;
    font-weight: 700;
    color: ${(props: AuthorProps) => (props.current ? '#9FA600' : '#3e414c')};
`;

interface AuthorProps {
    current: boolean;
}

const ChatMessage = (props: { user: any; message: ChatMessageType }) => (
    <Message>
        <Author
            current={props.message.creator.id === props.user.id}
            title={new Date(props.message.created_at).toDateString()}
        >
            {props.message.creator.name}:
        </Author>
        <span>{props.message.message}</span>
    </Message>
);

export default ChatMessage;
