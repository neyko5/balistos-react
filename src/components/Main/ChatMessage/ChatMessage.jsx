import React from 'react';
import PropTypes from 'prop-types';

const ChatMessage = props => (
  <div className="message">
    <span
      title={new Date(props.message.createdAt).toLocaleString()}
      className={`author ${props.message.user.username === props.username ? 'green' : 'grey'}`}
    >{props.message.user.username}:</span><span>{props.message.message}</span>
  </div>
    );

ChatMessage.propTypes = {
  username: PropTypes.string,
  message: PropTypes.shape({
    user: PropTypes.shape({
      username: PropTypes.string.isRequired,
    }).isRequired,
    message: PropTypes.string.isRequired,
    createdAt: PropTypes.string.isRequired,
  }).isRequired,
};

ChatMessage.defaultProps = {
  username: undefined,
};

export default ChatMessage;
