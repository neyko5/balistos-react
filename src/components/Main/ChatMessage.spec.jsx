import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import 'jest-styled-components'


import ChatMessage, {Author} from './ChatMessage';

Enzyme.configure({ adapter: new Adapter()});



test('chats from current user should have author be displayed in green', () => {
  const chatMessage = mount(
    <ChatMessage
      username={'testuser'}
      message={
        {
          user: { username: 'testuser' },
          message: 'text',
          createdAt: '2017-05-11T18:18:36.000Z',
        }
      }
    />);
  expect(chatMessage.find(Author)).toHaveStyleRule('color', '#9FA600');
});

test('chats from other users chats should have author be displayed in default color', () => {
  const chatMessage = mount(
    <ChatMessage
      username={'testuser'}
      message={
        {
          user: { username: 'otheruser' },
          message: 'text',
          createdAt: '2017-05-11T18:18:36.000Z',
        }
      }
    />);
  expect(chatMessage.find(Author)).toHaveStyleRule('color', '#3e414c');
});

test('chats should be displayed in default color if user is logged out', () => {
  const chatMessage = mount(
    <ChatMessage
      message={
        {
          user: { username: 'otheruser' },
          message: 'text',
          createdAt: '2017-05-11T18:18:36.000Z',
        }
      }
    />);
  expect(chatMessage.find(Author)).toHaveStyleRule('color', '#3e414c');
});

test('chat should have title parameter set', () => {
  const chatMessage = mount(
    <ChatMessage
      username={'testuser'}
      message={
        {
          user: { username: 'otheruser' },
          message: 'text',
          createdAt: '2017-05-11T18:18:36.000Z',
        }
      }
    />);
  expect(true).toBe(true);  
  expect(chatMessage.find(Author).prop('title')).not.toBeFalsy();
});

test('should display authors username', () => {
  const username = 'authoruser';
  const chatMessage = shallow(
    <ChatMessage
      username={'testuser'}
      message={
        {
          user: { username },
          message: 'text',
          createdAt: '2017-05-11T18:18:36.000Z',
        }
      }
    />);
  expect(chatMessage.contains(username)).toBe(true);
});

test('should display message content', () => {
  const testMessage = 'test message';
  const chatMessage = shallow(
    <ChatMessage
      username={'testuser'}
      message={
        {
          user: { username: 'testuser' },
          message: testMessage,
          createdAt: '2017-05-11T18:18:36.000Z',
        }
      }
    />);
  expect(chatMessage.contains(testMessage)).toBe(true);
});

test('should render correctly', () => {
  const chatMessage = renderer.create(
    <ChatMessage
      username={'testuser'}
      message={
        {
          user: { username: 'testuser' },
          message: 'test message',
          createdAt: '2017-05-11T18:18:36.000Z',
        }
      }
    />);
  const tree = chatMessage.toJSON();
  expect(tree).toMatchSnapshot();
});
