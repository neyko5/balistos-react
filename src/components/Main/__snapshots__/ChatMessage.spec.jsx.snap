// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="sc-bdVaJa eNFPzl"
>
  <span
    className="sc-bwzfXH fVnEOv"
    title="Thu May 11 2017"
  >
    testuser
    :
  </span>
  <span>
    test message
  </span>
</div>
`;
